src/pages/components/toggle.tsx

```diff
@@ -14,15 +15,29 @@ import ComponentExample from 'components/ComponentExample';
 import { Heading, Text, Paragraph } from 'components/ResponsiveComponents';
 
 const Page: React.FC = (): React.ReactElement => {
-  const [isChecked, setChecked] = useState(false);
+  const [isButtonChecked, setButtonChecked] = useState(false);
+  const [isInputChecked, setInputChecked] = useState(false);
 
-  const getAriaLabelText = (t: Function): string =>
-    t(`toggle.state.${isChecked ? 'on' : 'off'}`, {
-      name: t('toggle.label'),
+  const getAriaLabelText = (t: Function, inputVariant = false): string => {
+    const componentVariantState = inputVariant
+      ? isInputChecked
+      : isButtonChecked;
+
+    const componentVariantLabel = inputVariant
+      ? 'toggle.label.input'
+      : 'toggle.label.button';
+
+    return t(`toggle.state.${componentVariantState ? 'on' : 'off'}`, {
+      name: t(componentVariantLabel),
     });
+  };
+
+  const handleButtonClick = (newState: boolean): void => {
+    setButtonChecked(newState);
+  };
 
-  const handleClick = (newState: boolean): void => {
-    setChecked(newState);
+  const handleInputClick = (newState: boolean): void => {
+    setInputChecked(newState);
   };
```

Comment:
Made separate handlers for both. 
What are you opinion for these kinds, should it be like getAriaLabelText()-having same function for both but logic based on param?

Reply:
As long as same approach is used, my opinion is that it does not matter if there is one handler or separate handlers, since the handlers cannot be completely generic anyways.

Reply:
Separate handlers for both were added and it seems now more readable.